solr/core/src/java/org/apache/solr/util/SolrPropertyAnnotationInspector.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.util;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.AnnotationIntrospector;
+import com.fasterxml.jackson.databind.PropertyName;
+import com.fasterxml.jackson.databind.introspect.Annotated;
+import com.fasterxml.jackson.databind.introspect.AnnotatedField;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMember;
+import com.fasterxml.jackson.databind.introspect.AnnotatedMethod;
+import com.fasterxml.jackson.databind.util.BeanUtil;
+import org.apache.solr.common.annotation.Property;
+import org.apache.solr.search.SolrCache;
+//this class provides a maping between jackson's JsonProperty Annotation to Solr's Property annotation
```

Comment:
Use actual Javadoc comments please.  You should say why we're even doing this without requiring the reader to read SOLR-13841 (though referring to the issue is fine for another level of detail).  For example I think it's important to say here that we don't want SolrJ to depend on Jackson-annotations, and thus we duplicate the functionality for one annotation we've replicated.  A one-liner reference for how to install this to an ObjectMapper is helpful.

The current name isn't bad but I suggest naming this with Jackson in the name, like "SolrJacksonAnnotationInspector" to very clearly show this is Jackson related.